File: ehitchhiking-ui/src/app/shared/models/car.ts
export class Car {
	id: string;
	model: string;
	color: string;
	number: string;

	constructor(id: string, model: string = '', color: string = '', number: string = '') {
		this.id = id;
		this.model = model;
		this.color = color;
		this.number = number;
	}

	isEmpty(): boolean {
		return !this.model && !this.color && !this.number && !this.id;
	}

	changeData(model: string, color: string, number: string): void {
		this.model = model || this.model;
		this.color = color || this.color;
		this.number = number || this.number;
	}
}

File: ehitchhiking-api/src/main/java/com/exadel/ehitchhiking/Models/Car.java
package com.exadel.ehitchhiking.Models;


import lombok.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Set;


@Entity
@Table(name = "car")
@ToString
@EqualsAndHashCode
@NoArgsConstructor
public class Car {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Setter(AccessLevel.PRIVATE)
    @Getter
    @Column(name = "car_id")
    private Integer id;


    @Setter
    @Getter
    @Column(name = "car_color")
    private String color;


    @Setter
    @Getter
    @Column(name = "veh_number")
    private String number;


    @Setter
    @Getter
    @Column(name = "model")
    private String model;

    @Setter
    @Getter
    @ManyToOne
    @JoinColumn(name = "id_driver")
    private Driver driver;

    public Car(String color, String number, String model,
               Driver driver) {
        this.driver = driver;
        this.color = color;
        this.number = number;
        this.model = model;
    }
}
